helm-charts/ironic/ironic-config/dnsmasq.conf.j2

```diff
@@ -31,6 +31,15 @@ dhcp-boot=tag:efi,tag:!ipxe,snponly.efi
 dhcp-boot=/undionly.kpxe,{{ env.IRONIC_IP }}
 {% endif %}
 
+{%- if env.DHCP_HOSTS | length %}
+{% for mac in (env.DHCP_HOSTS | default('#') | replace('-', ':') | replace('\\,', ',')).split(',') -%}
+{%- if mac | length %}
+dhcp-host={{ mac }}
+{% endif %}
+{% endfor %}
+{% endif %}
+dhcp-ignore=tag:!known
```

Comment:
This one needs to be inside the if block so you may want to swap line 40 and line 41.